dataverse/orgsvc/C#/RemoveUnrefrencedQueues/RemoveUnrefrencedQueues/RemoveUnrefrencedQueues.cs

```
@@ -0,0 +1,207 @@
+
+                    if (typemask >= 1 && typemask <= 4 && activityParty.GetAttributeValue<EntityReference>("partyid").LogicalName == "queue")
+                    {
+                        tracingService.Trace("RemoveUnrefrencedQueues.GetQueuesToRemove: Applicable Queue: " + activityParty.GetAttributeValue<EntityReference>("partyid").Id.ToString());
+
+                        ConditionExpression condition = new ConditionExpression("msdyn_queueid", ConditionOperator.Equal, activityParty.GetAttributeValue<EntityReference>("partyid").Id);
+                        queues.Conditions.Add(condition);
+                    }
+                    if (typemask == 13)
+                    {
+                        tracingService.Trace("RemoveUnrefrencedQueues.GetQueuesToRemove: Applicable Entity: " + activityParty.GetAttributeValue<EntityReference>("partyid").Id.ToString());
+
+                        ConditionExpression condition = new ConditionExpression("msdyn_createdentityid", ConditionOperator.Equal, activityParty.GetAttributeValue<EntityReference>("partyid").Id.ToString());
+                        createdEntities.Conditions.Add(condition);
+                    }
+
+                }
+            }
+
+            // Put all the query conditions together
+            query.Criteria.AddFilter(queues);
+            query.Criteria.AddFilter(createdEntities);
+
+            tracingService.Trace("RemoveUnrefrencedQueues.GetQueuesToRemove: Executing Retrival");
+
+            return service.RetrieveMultiple(query);
+        }
+
+
+        /// <summary>
+        /// Execute method that is required by the IPlugin interface.
+        /// </summary>
+        /// <param name="serviceProvider">The service provider from which you can obtain the
+        /// tracing service, plug-in execution context, organization service, and more.</param>
+        public void Execute(IServiceProvider serviceProvider)
```

I don't see any error handling in this code.

As a sample, it should demonstrate this best practice and throw an InvalidPluginExecutionException even in the event of an unexpected error.